Source/Core/Core/AchievementManager.h

```diff
@@ -0,0 +1,47 @@
+// Copyright 2023 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#ifdef USE_RETRO_ACHIEVEMENTS
+#include <mutex>
+#include <string>
+#include <thread>
+
+#include <rcheevos/include/rc_api_user.h>
+#include <rcheevos/include/rc_runtime.h>
+
+#include "Common/Event.h"
+
+class AchievementManager
+{
+public:
+  static AchievementManager* GetInstance();
+  void Init();
+  void RequestPost();
+  void SetPassword(const std::string& password);
+
+private:
+  AchievementManager(){};
+
+  void ThreadProc();
+
+  void Login(const std::string& password);
+
+  void Logout();
+
+  template <typename RcRequest, typename RcResponse>
+  static void Request(RcRequest rc_request, RcResponse* rc_response,
+                      int (*init_request)(rc_api_request_t* request, const RcRequest* api_params),
+                      int (*process_response)(RcResponse* response, const char* server_response));
+
+  rc_runtime_t m_runtime;
```

Comment:
```suggestion
  rc_runtime_t m_runtime{};
```

so that this is guaranteed to have a deterministic state from the point of construction